webhook/strict.go

```diff
@@ -0,0 +1,313 @@
+						Message: "deprecated field set",
+						Paths:   []string{f.Name},
+					})
+					continue
+				}
+			}
+
+			switch v.Kind() {
+			case reflect.Ptr:
+				if v.IsNil() {
+					continue
+				}
+				errs = errs.Also(
+					strictValidateCreateReflected(v.Elem()).ViaField(f.Name),
+				)
+
+			case reflect.Struct:
+				errs = errs.Also(
+					strictValidateCreateReflected(v).ViaField(f.Name),
+				)
+
+			case reflect.Slice, reflect.Array:
+				for index := 0; index < v.Len(); index++ {
+					value := v.Index(index)
+					errs = errs.Also(
+						strictValidateCreateReflected(value).ViaFieldIndex(f.Name, index),
+					)
+				}
+
+			case reflect.Map:
+				it := v.MapRange()
+				for it.Next() {
+					key := it.Key()
+					value := it.Value()
+					errs = errs.Also(
+						strictValidateCreateReflected(value).ViaFieldKey(f.Name, key.String()),
+					)
+				}
+			}
+		}
+	}
+	return errs
+}
+
+func strictValidateUpdateReflectedByName(old, res interface{}, name string) *apis.FieldError {
+	oldValue := reflect.Indirect(reflect.ValueOf(old))
+	resValue := reflect.Indirect(reflect.ValueOf(res))
+
+	// If res is not a struct, don't even try to use it.
+	if resValue.Kind() != reflect.Struct {
+		return nil
+	}
+
+	// If old is not a struct, don't even try to use it.
+	if oldValue.Kind() != reflect.Struct {
+		return nil
+	}
+
+	resField := resValue.FieldByName(name)
+	if !resField.IsValid() || resField.Kind() != reflect.Struct {
+		return &apis.FieldError{
+			Message: "failed to inspect new object",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	oldField := oldValue.FieldByName(name)
+	if !oldField.IsValid() || oldField.Kind() != reflect.Struct {
+		return &apis.FieldError{
+			Message: "failed to inspect old object",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	if oldField.Type() != resField.Type() {
+		return &apis.FieldError{
+			Message: "failed to match filed type",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	return strictValidateUpdateReflected(oldField, resField)
+}
+
+func strictValidateUpdateReflected(oldField, resField reflect.Value) *apis.FieldError {
+	if !resField.IsValid() || resField.Kind() != reflect.Struct {
+		return &apis.FieldError{
+			Message: "failed to inspect new object",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	if !oldField.IsValid() || oldField.Kind() != reflect.Struct {
+		return &apis.FieldError{
+			Message: "failed to inspect old object",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	if oldField.Type() != resField.Type() {
+		return &apis.FieldError{
+			Message: "failed to match filed type",
+			Paths:   []string{apis.CurrentField},
+		}
+	}
+
+	var errs *apis.FieldError
+
+	// for each field,
+	for i := 0; i < resField.NumField(); i++ {
+		f := resField.Type().Field(i)
+		if v := resField.Field(i); v.IsValid() {
+			if o := oldField.Field(i); o.IsValid() {
+				if strings.HasPrefix(f.Name, "Deprecated") {
+					if differ(v, o) {
```

Comment:
This doesn't sound right. Since if `new` is empty and `old` wasn't it will return error right here.

Reply:
I am going to update this code. It is confusing....